jwst/ramp_fitting/ramp_fit.py

```diff
@@ -1502,12 +1551,17 @@ def fit_next_segment(start, end_st, end_heads, pixel_done, data_sect, mask_2d,
     #    - set number of end to 0
     #    - add slopes and variances to running sums
     #    - set pixel_done to True to designate all fitting done
+    #  For segments of this type, the final good group in the segment is a CR
+    #    and/or SAT and is not the final group in the ramp, and the variable
```

Comment:
Can the final group in a segment be a CR?  Why wouldn't that start a new segment?  Also, how can a group be good if it's either affected by a CR or is saturated?